spec/services/teachers/school_transfers/history_spec.rb

```diff
@@ -0,0 +1,355 @@
+        expect(transfer.joining_training_period).to eq(@training_period2)
+      end
+    end
+
+    context "when a teacher leaves school-led training at one school " \
+            "and joins provider-led training at another school" do
+      let(:lead_provider1) { FactoryBot.create(:lead_provider) }
+
+      before do
+        school_period1 = create_school_period(from: 3.years.ago, to: 2.years.ago)
+        @training_period1 = add_training_period(school_period1, from: 3.years.ago, to: 2.years.ago, programme_type: :school_led)
+        school_period2 = create_school_period(from: 2.years.ago)
+        @training_period2 = add_training_period(school_period2, from: 2.years.ago, programme_type: :provider_led, with: lead_provider1)
+      end
+
+      it "returns a new_provider transfer for lead provider #1" do
+        history = described_class.new(
+          school_periods: teacher.ect_at_school_periods,
+          lead_provider: lead_provider1
+        )
+
+        expect(history.transfers.size).to eq(1)
+        transfer = history.transfers.first
+        expect(transfer).to be_a(Teachers::SchoolTransfers::Transfer)
+        expect(transfer.type).to eq(:new_provider)
+        expect(transfer.status).to eq(:complete)
+        expect(transfer.teacher_type).to eq(:ect)
+        expect(transfer.leaving_training_period).to eq(@training_period1)
+        expect(transfer.joining_training_period).to eq(@training_period2)
+      end
+    end
+
+    context "when a teacher leaves school-led training at one school " \
+            "and joins school-led training at another school" do
+      let(:ect_at_school_period1) do
+        FactoryBot.create(
+          :ect_at_school_period,
+          started_on: 3.years.ago,
+          finished_on: 2.years.ago,
+          teacher:
+        )
+      end
+      let(:ect_at_school_period2) do
+        FactoryBot.create(
+          :ect_at_school_period,
+          :ongoing,
+          started_on: 2.years.ago,
+          teacher:
+        )
+      end
+
+      let(:lead_provider1) { FactoryBot.create(:lead_provider) }
+
+      before do
+        school_period1 = create_school_period(from: 3.years.ago, to: 2.years.ago)
+        add_training_period(school_period1, from: 3.years.ago, to: 2.years.ago, programme_type: :school_led)
+        school_period2 = create_school_period(from: 2.years.ago)
+        add_training_period(school_period2, from: 2.years.ago, to: 1.year.ago, programme_type: :school_led)
+        add_training_period(school_period2, from: 1.year.ago, programme_type: :provider_led, with: lead_provider1)
+      end
+
+      it "returns no transfers for lead provider #1" do
+        history = described_class.new(
+          school_periods: teacher.ect_at_school_periods,
+          lead_provider: lead_provider1
+        )
+
+        expect(history.transfers).to be_empty
+      end
+    end
+  end
+
+private
+
+  def create_school_period(from:, to: nil)
```

Comment:
Hi @cpjmcquillan how about extracting those methods into helpers? like we did in `spec/support/mentorship_period_helpers.rb` and `spec/support/school_partnership_helpers.rb`

Reply:
I think I'm going to leave them defined here for now. 

But if they end up being useful in other transfer-related tests, I'll definitely extract them 👌🏼